public/utils/pipeline/operator.ts

```diff
@@ -0,0 +1,8 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+export abstract class Operator<T, P> {
+  abstract execute(v: T): Promise<P>;
```

Comment:
do we need context parameter? and how about name it as `Task`

Reply:
> how about name it as Task

Sounds good, I like the name, let me update.

> do we need context parameter?

Is there an example?

Reply:
I don't have a real use case, but this could helpful if we have context for tasks share intermediate result during execution. For example, task 3 want to use a output of task 1

Reply:
The current design is each task will add extra properties to its output, and the output will become the next task's input. But I got your point which makes to me, I think we can tweaks the implementation when there is a real use case.